tokendito/okta.py

```diff
@@ -303,26 +309,42 @@ def get_session_token(config, primary_auth, headers):
     return session_token
 
 
-def get_oauth2_token(authz_code_flow_data, authorize_code):
+def get_access_token(oauth2_config, oauth2_session_data, authorize_code):
     """Get OAuth token from Okta by calling /token endpoint.
 
+    This method does not seem to be needed, calling /authorize sets the idx cookies,
+    but we put it here to follow the flow vervatim.
+
     :param url: URL of the Okta OAuth token endpoint
     :return: OAuth token
     """
-    payload = {
-        "code": authorize_code,
-        "state": authz_code_flow_data["state"],
-        "grant_type": authz_code_flow_data["grant_type"],
-        "redirect_uri": authz_code_flow_data["redirect_uri"],
-        "client_id": authz_code_flow_data["client_id"],
-        "code_verifier": authz_code_flow_data["code_verifier"],
-    }
+    try:
+        payload = {
+            "code": authorize_code,
+            "state": oauth2_session_data["state"],
+            "grant_type": oauth2_session_data["grant_type"],
+            "redirect_uri": oauth2_session_data["redirect_uri"],
+            "client_id": oauth2_config["client_id"],
+            "code_verifier": oauth2_session_data["code_verifier"],
+        }
+    except KeyError as e:
+        logger.error(f"Missing key in config:{e}")
+        sys.exit(1)
+
     headers = {"accept": "application/json"}
     # Using the http_client to make the POST request
     response = HTTP_client.post(
-        authz_code_flow_data["token_endpoint_url"], data=payload, headers=headers, return_json=True
+        oauth2_config["token_endpoint"], data=payload, headers=headers, return_json=True
     )
-    return response
+    # We now have response['access_token'] and response['id_token'], but we dont seem to need
+    # them to access the resources.
+    access_token = None
+    try:
+        access_token = response["access_token"]
+    except KeyError:
+        logger.debug(f"Error parsing response: {json.dumps(response)}")
+        # Don't do anything but a debug message, as the /token call doesnt seem to be needed.
+    return access_token
 
 
 def get_client_id(config):
```

Comment:
The code to call the /token endpoint is there only because it is in the flow, we do get tokens but we dont do anything with them, as soon as we call /authorize we get the idx cookies.  We can include in a future PR once we figure out where it is needed.